src/internet_identity/tests/integration/openid.rs

```diff
@@ -51,12 +52,176 @@ fn can_link_google_account() -> Result<(), RejectResponse> {
     Ok(())
 }
 
+/// Verifies that Microsoft Accounts can be added
+#[test]
+fn can_link_microsoft_account() -> Result<(), RejectResponse> {
+    let env = env();
+    let canister_id = setup_canister(&env);
+    let (jwt, salt, _claims, test_time, test_principal, test_authn_method) =
+        openid_microsoft_test_data();
+
+    let identity_number = create_identity_with_authn_method(&env, canister_id, &test_authn_method);
+
+    let time_to_advance = Duration::from_millis(test_time) - Duration::from_nanos(time(&env));
+    env.advance_time(time_to_advance);
+
+    assert_eq!(
+        number_of_openid_credentials(&env, canister_id, test_principal, identity_number)?,
+        0
+    );
+
+    let _ = api::openid_credential_add(
+        &env,
+        canister_id,
+        test_principal,
+        identity_number,
+        &jwt,
+        &salt,
+    )?;
+
+    assert_eq!(
+        number_of_openid_credentials(&env, canister_id, test_principal, identity_number)?,
+        1
+    );
+
+    Ok(())
+}
+
+/// Verifies that the same Microsoft account cannot be linked to two different identities
+#[test]
+fn cannot_link_same_microsoft_account_to_two_identities() -> Result<(), RejectResponse> {
+    let env = env();
+    let canister_id = setup_canister(&env);
+    let (jwt, salt, _claims, test_time, test_principal, test_authn_method) =
+        openid_microsoft_test_data();
+    let (jwt2, salt2, _claims2, test_time2, test_principal2, test_authn_method2) =
+        openid_microsoft_test_data_same_microsoft_different_user();
+
+    let identity_number = create_identity_with_authn_method(&env, canister_id, &test_authn_method);
+    let identity_number2 =
+        create_identity_with_authn_method(&env, canister_id, &test_authn_method2);
+
+    let time_to_advance = Duration::from_millis(test_time) - Duration::from_nanos(time(&env));
```

Comment:
Nit. Seems like a common pattern, could be lifted to a helper function, e.g., `sync_test(&env, test_time)`